File: base-class-module/src/main/java/jingya/com/base_class_module/ResultCallBack.java
package jingya.com.base_class_module;

/**
 * @author Kuky
 *         <p>
 *         网络数据请求回调
 */
public class ResultCallBack {

    public interface OnHttpRequestCallBack<T> {
        void onSuccess(T t);

        void onFail(String msg);
    }

    public interface onListCallBack<T> {
        void onSucceed(T t);

        void onEmpty();

        void onFailed();
    }

    public interface OnResultBack<T> {
        void onResult(T t);
    }
}

File: base-kt-module/src/main/java/com/kuky/base_kt_module/ResultCallBack.kt
package com.kuky.base_kt_module

/**
 * @author kuky
 * @description 网络数据请求回调
 */
class ResultCallBack {

    interface OnHttpRequestCallBack<T> {
        fun onSuccess(t: T)

        fun onFail(msg: String)
    }

    interface onListCallBack<T> {
        fun onSucceed(t: T)

        fun onEmpty()

        fun onFailed()
    }

    interface OnResultBack<T> {
        fun onResult(t: T)
    }
}
